src/libre/ast.c

```diff
@@ -912,3 +913,128 @@ ast_node_type_name(enum ast_expr_type t)
 	case AST_EXPR_TOMBSTONE: return "TOMBSTONE";
 	}
 }
+
+/*
+ * TODO: we could use this for AST->NFA conversion and decide to call
+ * re_strings() instead (also for a set of alts of strings)
+ *
+ * Return values:
```

Comment:
Is it worth making this an enum? `ast_is_literal` sounds like a function that would return bool in a >= C99 context, but if someone just checks `if (ast_is_literal(...)) { ...}` they will end up with surprises from the error case. That feels like an easy mistake to make, and since this is in the public API it will be a hassle to change later.